python/lsst/pipe/tasks/drpAssociationPipe.py

```diff
@@ -74,6 +123,13 @@ class DrpAssociationPipeConnections(pipeBase.PipelineTaskConnections,
         dimensions=("tract", "patch"),
     )
 
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if not config.doSolarSystemAssociation:
+            self.inputs.remove("solarSystemObjectTables")
+            self.outputs.remove("associatedSsSources")
```

Comment:
Should `unassociatedSsObjects` be removed here, too?

I'd also recommend slightly different syntax for this, just because it's a bit easier to read (what you have is not incorrect):
```suggestion
            del self.associatedSsSources
```

Reply:
Had I used your syntax, I would have noticed that I renamed `solarSystemObjectTables` into `ssObjectTableRefs`. Changed!